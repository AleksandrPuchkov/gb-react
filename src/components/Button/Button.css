.submitbutton {
    padding: 5px;
    height: 30px;
}

.submitbutton:hover {cursor: pointer}