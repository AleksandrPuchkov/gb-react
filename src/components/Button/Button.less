.submitbutton {
    padding: 5px;
    height: 30px;

    &:hover {
        cursor: pointer
    }
}