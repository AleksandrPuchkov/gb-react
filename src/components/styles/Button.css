.inputButton {
    width: 200px;
    height: 40px;
    border-radius: 5px;
    border: 0px;
    background-color: #999;
    color: white;
    font-weight: 700;
    transition: background-color 0.3s ease;
}

.inputButton:hover {
    background-color: #555;
    cursor: pointer;
}