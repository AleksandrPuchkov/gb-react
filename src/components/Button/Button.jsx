import React from 'react';
import './Button.css';

export const Button = ({ disabled }) => (
  <button
    className="submitbutton"
    data-testid="submitbutton"
    type="submit"
    disabled={disabled}
  >
    Отправить
  </button>
);
