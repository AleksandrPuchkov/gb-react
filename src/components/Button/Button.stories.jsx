import React from 'react';

import { Button } from './Button';

export default {
  title: 'SubmitButton',
  component: Button,
  argTypes: {
    onClick: { action: 'click' },
  },
};

const Template = (args) => <Button {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  primary: true,
  disabled: false,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};

export const RedButton = Template.bind({});
RedButton.args = {
  color: 'red',
};
