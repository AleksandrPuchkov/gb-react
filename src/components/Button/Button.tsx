import React, { FC, memo } from 'react';
import './Button.less';

interface ButtonProps {
  disabled: boolean,
}

export const Button: FC<ButtonProps> = memo(({ disabled }) =>
(
  <button
    className="submitbutton"
    data-testid="submitbutton"
    type="submit"
    disabled={disabled}
  >
    Отправить
  </button>
)
);
