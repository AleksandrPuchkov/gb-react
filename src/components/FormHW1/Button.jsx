import './styles/Button.css'

export const Button = (props) => {
    return <button onClick={props.click}  className="inputButton">Изменить Message</button>
}