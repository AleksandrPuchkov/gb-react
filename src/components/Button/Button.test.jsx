import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';

describe('Button', () => {
  it('render component', () => {
    render(<Button />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Button />);
    expect(screen.getByText(/Отправить/)).toBeInTheDocument;
  });

  it('render component with button role', () => {
    render(<Button />);
    expect(screen.getByTestId('submitbutton')).toBeInTheDocument;
  });

  it('Button is active', () => {
    render(<Button />);
    expect(screen.getByTestId('submitbutton')).not.toBeDisabled;
  });
});
