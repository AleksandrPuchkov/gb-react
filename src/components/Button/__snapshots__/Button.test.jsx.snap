// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="submitbutton"
    data-testid="submitbutton"
    type="submit"
  >
    Отправить
  </button>
</DocumentFragment>
`;
